src/components/application_manager/src/application_manager_impl.cc

```diff
@@ -1526,6 +1538,51 @@ void ApplicationManagerImpl::OnServiceEndedCallback(
   }
 }
 
+bool ApplicationManagerImpl::ShouldAddAppIDForService(
+    hmi_apis::Common_ServiceType::eType service_type,
+    hmi_apis::Common_ServiceEvent::eType service_event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  if (hmi_apis::Common_ServiceType::RPC == service_type &&
+      !is_first_rpc_service_accepted_ &&
+      hmi_apis::Common_ServiceEvent::REQUEST_ACCEPTED == service_event) {
+    is_first_rpc_service_accepted_ = true;
+    return false;
+  } else {
+    return true;
+  }
+}
+
+void ApplicationManagerImpl::ProcessServiceStatusUpdate(
+    const uint32_t connection_key,
+    hmi_apis::Common_ServiceType::eType service_type,
+    hmi_apis::Common_ServiceEvent::eType service_event,
+    hmi_apis::Common_ServiceUpdateReason::eType service_update_reason) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  LOG4CXX_DEBUG(logger_,
+                "Processing status update with connection key: "
+                    << connection_key << " service type: " << service_type
+                    << " service_event " << service_event
+                    << "service_update_reason " << service_update_reason);
+
+  auto notification =
+      MessageHelper::CreateOnServiceStatusUpdateNotification(service_type);
+  (*notification)[strings::msg_params][hmi_notification::service_event] =
+      service_event;
+
+  if (ShouldAddAppIDForService(service_type, service_event)) {
```

Comment:
@mked-luxoft looks like you adding app_id optionally according to following part of the proposal 
```
Note that the only time when SDL would not be able provide the appID would be during the first StartService request for the RPC service before RAI was sent.
```
The point there is that SDL is **unable** to provide app_id in case applicaiton start secure service before RAI.

So in the code in that place you probably have no ability to get the connection_key. Does it mean that `connection_key` arg is broken? or NULL? I would better make this argument optional in the ProcessServiceStatusUpdate function, and find out place where you can't pass it.